src/ice-engine/threadpool.cpp

```diff
@@ -0,0 +1,58 @@
+//#include "common.h"
+#include "ice-engine/threadpool.h"
+#include <functional>
+#include <iostream>
+#include <future>
+#include <thread>
+#include <queue>
+#include <vector>
+#include <condition_variable>
+
+threadpool::threadpool(uint8_t num_threads)
+{
+    //
+    worker_threads.reserve(num_threads);
+    // initialize to be running a worker task
+    // worker task should simply use cond variable to access whether or not there is work
+    int i = 0;
+    mStop = false;
+    while(worker_threads.size() < worker_threads.capacity())
+    {
+        worker_threads.push_back(std::thread([this]
+        {
+           this->run_worker_thread();
+	}));
+	worker_threads[i].detach();
+	++i;
+    }
+}
+
+
+void threadpool::run_worker_thread(void)
+{
+    while(1)
```

Comment:
You already have a member variable for stop which would signal the end of the loop, rather than running an infinite loop and breaking, you could just exit the loop when `mStop` is set to `true`.